src/components/implementation/no_interface/scheddev/sl_lock.c

```diff
@@ -0,0 +1,67 @@
+#include "sl.h"
+#include "sl_lock.h"
+
+void
+sl_lock_init(struct sl_lock *lock)
+{
+    assert(lock);
+    *lock = SL_LOCK_STATIC_INIT();
+}
+
+thdid_t
+sl_lock_holder(struct sl_lock *lock)
+{
+    assert(lock);
+    return lock->holder;
+}
+
+
+void
+sl_lock_lock(struct sl_lock *lock)
```

Comment:
There are already 80 comments on this PR. I'm sorry I forgot to reply to the previous comment, and if it is mandatory for inclusion I will change the names of the functions.